torch/_inductor/ir.py

```diff
@@ -1252,8 +1253,21 @@ def _dynamic_reshape_indexer(old_size, new_size):
         size_hint = V.graph.sizevars.size_hint
         vars = [sympy_symbol(f"view{i}") for i in range(len(new_size))]
 
-        stack_new = list(zip(vars, new_size))
-        stack_old = list(old_size)
+        def any0(l):
+            return any([size_hint(x) for x in list(l)])
```

Comment:
this looks like any_non_zero? On a higher level, should we get rid of zero-sized tensors before we come to actual reindexers as handling those cases here seems much trickier, as you note sympy_symbol is assumed to be positive. 
cc @chillee who last touched sympy_symbol

Reply:
Yeah, i think everything should be non-zero already?

Reply:
Apparently no, there's viewing one 0-element tensor as another one, but my suggestion is dealing with this situation before we come to reindexing.

Reply:
>  On a higher level, should we get rid of zero-sized tensors before we come to actual reindexers 

@Chillee @ngimel 
could you guys elaborate? Does this mean `torchdynamo` should be graph breaking or if there should be some alternative path for lowering zero-dim tensors?